p2p/exchange.go

```diff
@@ -121,30 +126,56 @@ func (ex *Exchange[H]) Head(ctx context.Context) (H, error) {
 		defer cancel()
 	}
 
+	reqParams := header.DefaultHeadParams()
+	for _, opt := range opts {
+		opt(&reqParams)
+	}
+
+	peers := ex.peerTracker.getPeers()
+	if reqParams.DisableSubjectiveInit != nil && len(peers) >= numUntrustedHeadRequests {
+		// subjective initialisation was disabled, request head from a subset of
+		// tracked peers
+		peers = peers[:numUntrustedHeadRequests]
+	} else {
+		// default to using trusted peers
+		peers = ex.trustedPeers()
+	}
+
 	var (
-		zero         H
-		trustedPeers = ex.trustedPeers()
-		headerRespCh = make(chan H, len(trustedPeers))
-		headerReq    = &p2p_pb.HeaderRequest{
+		zero      H
+		headerReq = &p2p_pb.HeaderRequest{
 			Data:   &p2p_pb.HeaderRequest_Origin{Origin: uint64(0)},
 			Amount: 1,
 		}
+		headerRespCh = make(chan H, len(peers))
 	)
-	for _, from := range trustedPeers {
+	for _, from := range peers {
 		go func(from peer.ID) {
 			headers, err := ex.request(reqCtx, from, headerReq)
 			if err != nil {
 				log.Errorw("head request to trusted peer failed", "trustedPeer", from, "err", err)
 				headerRespCh <- zero
 				return
 			}
+			// if tracked (untrusted) peers were requested, verify head
+			if reqParams.DisableSubjectiveInit != nil {
+				err = reqParams.DisableSubjectiveInit.Verify(headers[0])
```

Comment:
At this point, I really don't like the name `DisableSubjectiveInit`. It's counter intuitive for me. (but I don't have better alternatives yet)

Reply:
For example, trusted hash is used for "subjective initialization". So "we (subjectively) know/trust a given hash" == "subjective initialization".
In here "DisableSubjectiveInit" means "do not depend on trusted peers, verify against given header". For me this is very similar to passing trusted hash (but there's a header, not a hash, that we subjectively trust). But the option name is **`Disable`**`SubjectiveInit`.

Reply:
Okay how about we rename it to `WithTrustedHead(subjectiveHead header.Header)` 

This would indicate to the exchange that we already have a trusted head so it can use the untrusted peers and verify the network head against the subjective head.

Reply:
I like this proposal!